queries/perl/highlights.scm

```diff
@@ -1,85 +1,116 @@
+; Misc keywords
+[
+  "my" "our" "local"
+  "next" "last" "redo"
+  "goto"
+  "package"
+;  "do"
+;  "eval"
+] @keyword
+
+; Keywords for including
+[ "use" "no" "require" ] @include
+
+; Keywords that mark conditional statements
+[ "if" "elsif" "unless" "else" ] @conditional
+
+; Keywords that mark repeating loops
+[ "while" "until" "for" "foreach" ] @repeat
+
+; Keyword for return expressions
+[ "return" ] @keyword.return
+
+; Keywords for phaser blocks
+; TODO: Ideally these would be @keyword.phaser but vim-treesitter doesn't
+;   have such a thing yet
+[ "BEGIN" "CHECK" "UNITCHECK" "INIT" "END" ] @keyword.function
+
+; Keywords to define a function
+[ "sub" ] @keyword.function
+
+; Lots of builtin functions, except tree-sitter-perl doesn't emit most of
+;   these yet
+;[
+;  "print" "printf" "sprintf" "say"
+;  "push" "pop" "shift" "unshift" "splice"
+;  "exists" "delete" "keys" "values"
+;  "each"
+;] @function.builtin
```

Comment:
Isn't it possible to recognize them with `#any-of?`? I'd imagine that they are at least present in the tree as `identifier`s in call position.

Reply:
Uh quite possibly? :) I have never seen/heard-of `#any-of` before so perhaps a reference link would be useful?

Reply:
You can look at other queries (Python, cmake). The documentation is `:h lua-treesitter-predicates`.

Reply:
Ah, I see. That'd add quite a lot. To go through the *full* set of core functions, would probably add about 200 names here. I'm happy to do that but perhaps in a separate commit, to keep this one moving.
What do people think?

Reply:
sounds good. I feel that the distinction built-in function vs regular one is not essential